packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -527,7 +530,11 @@ export class CfnParser {
         return { Condition: condition.logicalId };
       }
       default:
-        throw new Error(`Unsupported CloudFormation function '${key}'`);
+        if (this.options.context === CfnParsingContext.RULES) {
+          return this.handleRulesIntrinsic(key, object);
```

Comment:
It feels a bit odd to do this here in the `default` block. Won't that mean that we will (incorrectly) successfully parse Rules that make use of unsupported intrinsics like `Fn::Split`; or have I misunderstood the docs, and those intrinsics can be used _in addition to_ the rule-specific ones?

Reply:
Yes. That was I was trying to explain with the poetic comment above, but obviously I failed :).

Basically, the point of doing it this way is not to hard-code the list of acceptable functions that Rules support. So yes, in the current form, if you used `Fn::Split` in your Rule, we would not catch it.

Does this make sense? Or are you saying we should hard-code the list of supported functions here?

Reply:
The only reason to hard-code the list of functions would be to get runtime (vs deploy-time) errors when the template is invalid, at the potential cost of having to maintain the list and keep it up to date. I think it's relatively minor, so happy with it as-is if you are.